Keycloak admin client fail with "authHeader" is null when using classic extensions

### Describe the bug

Here is a previous issue which seems to be fixed and closed, could be related: https://github.com/quarkusio/quarkus/issues/30089.

When I inject org.keycloak.admin.client.Keycloak and try to call some method (e.g. keycloak.serverInfo().getInfo()) it is giving me the following error:
```
Caused by: java.lang.NullPointerException: Cannot invoke "String.startsWith(String)" because "authHeader" is null
        at org.keycloak.admin.client.resource.BearerAuthFilter.filter(BearerAuthFilter.java:53)
        at org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.filterRequest(ClientInvocation.java:644)
```

It is only happening when using classic extensions, when using reactive extensions it is working fine.
For the test I'm using latest Quarkus (3.4.2) and latest LTS version (3.2.6.Final), with these extensions it is not working:
```
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-resteasy</artifactId>
</dependency>
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-resteasy-jsonb</artifactId>
</dependency>
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-keycloak-admin-client</artifactId>
</dependency>
```

But if I use reactive extensions it is working fine:
```
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-resteasy-reactive</artifactId>
</dependency>
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-resteasy-reactive-jsonb</artifactId>
</dependency>
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-keycloak-admin-client-reactive</artifactId>
</dependency>
```

### Expected behavior

Behaviour should be the same as when using reactive extensions.

### Actual behavior

It is throwing NPE error.

### How to Reproduce?

[code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/12874488/code-with-quarkus.zip)

Here is a minimum reproduction case (freshly generated Quarkus project).
In pom there are classic extensions and commented out reactive extension.
Also in the application properties there are basic properties for keycloak admin client.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

java version "1.8.0_381"
Java(TM) SE Runtime Environment (build 1.8.0_381-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.381-b09, mixed mode)

and

openjdk version "17.0.8" 2023-07-18
OpenJDK Runtime Environment Temurin-17.0.8+7 (build 17.0.8+7)
OpenJDK 64-Bit Server VM Temurin-17.0.8+7 (build 17.0.8+7, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.4.2 and 3.2.6.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

apache-maven-3.9.5

### Additional information

_No response_

--- a/extensions/keycloak-admin-client/deployment/src/main/java/io/quarkus/keycloak/adminclient/deployment/KeycloakAdminClientProcessor.java
+++ b/extensions/keycloak-admin-client/deployment/src/main/java/io/quarkus/keycloak/adminclient/deployment/KeycloakAdminClientProcessor.java
@@ -13,6 +13,8 @@ import org.keycloak.json.StringOrArrayDeserializer;
 import org.keycloak.json.StringOrArraySerializer;
 
 import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
+import io.quarkus.deployment.Capabilities;
+import io.quarkus.deployment.Capability;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.ExecutionTime;
@@ -49,8 +51,10 @@ public class KeycloakAdminClientProcessor {
     @Record(ExecutionTime.STATIC_INIT)
     @Produce(ServiceStartBuildItem.class)
     @BuildStep
-    public void integrate(ResteasyKeycloakAdminClientRecorder recorder, TlsConfig tlsConfig) {
-        recorder.setClientProvider(tlsConfig.trustAll);
+    public void integrate(ResteasyKeycloakAdminClientRecorder recorder, TlsConfig tlsConfig, Capabilities capabilities) {
+        boolean areJSONBProvidersPresent = capabilities.isPresent(Capability.RESTEASY_JSON_JSONB)
+                || capabilities.isPresent(Capability.RESTEASY_JSON_JSONB_CLIENT);
+        recorder.setClientProvider(tlsConfig.trustAll, areJSONBProvidersPresent);
     }
 
     @Record(ExecutionTime.RUNTIME_INIT)

--- a/extensions/keycloak-admin-client/runtime/src/main/java/io/quarkus/keycloak/adminclient/ResteasyKeycloakAdminClientRecorder.java
+++ b/extensions/keycloak-admin-client/runtime/src/main/java/io/quarkus/keycloak/adminclient/ResteasyKeycloakAdminClientRecorder.java
@@ -6,7 +6,10 @@ import java.util.function.Supplier;
 
 import javax.net.ssl.SSLContext;
 
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.Produces;
 import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.core.MediaType;
 
 import org.keycloak.admin.client.ClientBuilderWrapper;
 import org.keycloak.admin.client.Keycloak;
@@ -14,6 +17,7 @@ import org.keycloak.admin.client.KeycloakBuilder;
 import org.keycloak.admin.client.spi.ResteasyClientClassicProvider;
 
 import io.quarkus.keycloak.admin.client.common.KeycloakAdminClientConfig;
+import io.quarkus.resteasy.common.runtime.jackson.QuarkusJacksonSerializer;
 import io.quarkus.runtime.RuntimeValue;
 import io.quarkus.runtime.annotations.Recorder;
 
@@ -58,14 +62,25 @@ public class ResteasyKeycloakAdminClientRecorder {
         };
     }
 
-    public void setClientProvider(boolean tlsTrustAll) {
+    public void setClientProvider(boolean tlsTrustAll, boolean areJSONBProvidersPresent) {
         Keycloak.setClientProvider(new ResteasyClientClassicProvider() {
             @Override
             public Client newRestEasyClient(Object customJacksonProvider, SSLContext sslContext, boolean disableTrustManager) {
                 // point here is to use default Quarkus providers rather than org.keycloak.admin.client.JacksonProvider
                 // as it doesn't work properly in native mode
-                return ClientBuilderWrapper.create(sslContext, tlsTrustAll || disableTrustManager).build();
+                var builder = ClientBuilderWrapper.create(sslContext, tlsTrustAll || disableTrustManager);
+                if (areJSONBProvidersPresent) {
+                    // when both Jackson and JSONB providers are present, we need to ensure Jackson is used
+                    builder.register(new AppJsonQuarkusJacksonSerializer(), 100);
+                }
+                return builder.build();
             }
         });
+    }
+
+    // makes media type more specific which ensures that it will be used first
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    static class AppJsonQuarkusJacksonSerializer extends QuarkusJacksonSerializer {
     }
 }

